src/analysis/Environments/Environments.ts

```diff
@@ -187,13 +187,12 @@ export function PauseButton(props: PauseButtonProps): ReactNode {
   const shouldShowPauseButton =
     isPauseSupported(getToolLabelFromCloudEnv(cloudEnvironment)) &&
     currentUser === getCreatorForCompute(cloudEnvironment);
```

Comment:
More IAM/permissions logic baked into creator and scattered throughout UI 😬

Reply:
Out of curiosity, is there another recommended paradigm in our UI code to follow?